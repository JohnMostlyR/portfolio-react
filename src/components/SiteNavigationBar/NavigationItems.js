import React from 'react';
import PropTypes from 'prop-types';
import {FormattedMessage} from 'react-intl';

import StyledUL from './StyledUL';
import StyledLI from './StyledLI';
import SiteNavLink from './SiteNavLink';
import SiteNavigationIcon from './SiteNavigationIcon';
import Description from './Description';

const NavigationItems = (props, context) => {
  return (
    <StyledUL>
      <StyledLI>
        <SiteNavLink to={'/'} data-isactive={context.router.route.location.pathname === '/'}>
          <SiteNavigationIcon name={'home'} fixedWidth/>
          <Description>
            <FormattedMessage id='portfolio.nav.home'
                              defaultMessage='Intro'
            />
          </Description>
        </SiteNavLink>
      </StyledLI>
      <StyledLI>
        <SiteNavLink to={'/about'}
                     data-isactive={context.router.route.location.pathname === '/about'}>
          <SiteNavigationIcon name={'address-card'} fixedWidth/>
          <Description>
            <FormattedMessage id='portfolio.nav.about'
                              defaultMessage='About'
            />
          </Description>
        </SiteNavLink>
      </StyledLI>
      <StyledLI>
        <SiteNavLink to={'/skills'}
                     data-isactive={context.router.route.location.pathname === '/skills'}>
          <SiteNavigationIcon name={'code'} fixedWidth/>
          <Description>
            <FormattedMessage id='portfolio.nav.skills'
                              defaultMessage='Skills'
            />
          </Description>
        </SiteNavLink>
      </StyledLI>
      <StyledLI>
        <SiteNavLink to={'/projects'}
                     data-isactive={context.router.route.location.pathname === '/projects'}>
          <SiteNavigationIcon name={'list'} fixedWidth/>
          <Description>
            <FormattedMessage id='portfolio.nav.projects'
                              defaultMessage='Projects'
            />
          </Description>
        </SiteNavLink>
      </StyledLI>
      <StyledLI>
        <SiteNavLink to={'/contact'}
                     data-isactive={context.router.route.location.pathname === '/contact'}>
          <SiteNavigationIcon name={'send'} fixedWidth/>
          <Description>
            <FormattedMessage id='portfolio.nav.contact'
                              defaultMessage='Contact'
            />
          </Description>
        </SiteNavLink>
      </StyledLI>
    </StyledUL>
  )
};

NavigationItems.contextTypes = {
  router: PropTypes.object,
};

export default NavigationItems;
