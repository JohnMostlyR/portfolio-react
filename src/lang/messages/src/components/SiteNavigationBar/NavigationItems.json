[
  {
    "id": "portfolio.nav.home",
    "defaultMessage": "Intro"
  },
  {
    "id": "portfolio.nav.about",
    "defaultMessage": "About"
  },
  {
    "id": "portfolio.nav.skills",
    "defaultMessage": "Skills"
  },
  {
    "id": "portfolio.nav.projects",
    "defaultMessage": "Projects"
  },
  {
    "id": "portfolio.nav.contact",
    "defaultMessage": "Contact"
  }
]