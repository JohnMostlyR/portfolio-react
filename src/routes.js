import HomePage from './pages/home';
import AboutPage from './pages/about';
import SkillsPage from './pages/skills';
import ProjectsPage from './pages/projects';
import ContactPage from './pages/contact';

const routes = [
  {
    path: '/',
    exact: true,
    component: HomePage,
  },
  {
    path: '/about',
    component: AboutPage,
  },
  {
    path: '/skills',
    component: SkillsPage,
  },
  {
    path: '/projects',
    component: ProjectsPage,
  },
  {
    path: '/contact',
    component: ContactPage,
  },
];

export default routes;