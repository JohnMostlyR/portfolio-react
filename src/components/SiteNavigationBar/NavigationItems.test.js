import React from 'react';
import { shallow } from 'enzyme';
import createRouterContext from 'react-router-test-context';

import NavigationItems from './NavigationItems';
import SiteNavLink from './SiteNavLink';
import routes from '../../routes';

describe('NavigationItems', () => {
  const routerContext = createRouterContext();
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
        <NavigationItems/>,
        { context: routerContext },
    );
  });

  it('Should render and match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should have a `SiteNavLink` component for each route in the correct sequence', () => {
    routes.forEach((route, idx) => {
      expect(
          wrapper
              .find(SiteNavLink)
              .at(idx)
              .prop('to')
      ).toBe(route.path);
    });
  });
});