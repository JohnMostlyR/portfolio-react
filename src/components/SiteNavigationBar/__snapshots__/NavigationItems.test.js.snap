// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItems Should render and match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "function",
    "props": Object {
      "intl": Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {},
        "now": [Function],
        "textComponent": "span",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "intl": Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "formatDate": [Function],
          "formatHTMLMessage": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatPlural": [Function],
          "formatRelative": [Function],
          "formatTime": [Function],
          "formats": Object {},
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralFormat": [Function],
            "getRelativeFormat": [Function],
          },
          "locale": "en",
          "messages": Object {},
          "now": [Function],
          "textComponent": "span",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
